examples/typescript/lnd/src/nodeHandler.ts

```diff
@@ -0,0 +1,77 @@
+import { LnRpc, WalletUnlockerRpc } from '@radar/lnrpc';
+import inquirer from 'inquirer';
+
+enum BaseAnswers {
+  OpenChannel = 'Open a channel',
+  ListPeers = 'List peers',
+  ListPendingChannels = 'List pending channels',
+  ListOpenChannels = 'List open channels',
+}
+
+type Base = {
+  action: BaseAnswers;
+};
+
+export default class NodeHandler {
+  constructor(private lnrpc: LnRpc & WalletUnlockerRpc) {}
+
+  whatWouldYouLikeToDo = async () => {
+    const info = await this.lnrpc.getInfo();
+    const { action } = await inquirer.prompt<Base>({
+      type: 'list',
+      name: 'action',
+      message: `Hi ${info.alias}, what would you like to do?`,
+      choices: Object.values(BaseAnswers),
+    });
+    let answer = await this.getAnswer(action);
+    while (typeof answer !== 'string') {
+      answer = await answer();
+    }
+    console.log(answer);
+    await this.whatWouldYouLikeToDo();
+  };
+
+  getAnswer = async (
+    question: BaseAnswers,
+  ): Promise<string | (() => Promise<string>)> => {
+    switch (question) {
+      case 'Open a channel': {
+        return this.specifyChannelOpeningParameters;
+      }
+      case 'List peers': {
+        const { peers } = await this.lnrpc.listPeers();
+        return `Your peers:\n\n`.concat(peers.map(p => p.address).join('\n'));
+      }
+      case 'List pending channels': {
+        const info = await this.lnrpc.pendingChannels();
+        return `Pending channels:\n${JSON.stringify(info)}`;
+      }
+      case 'List open channels': {
+        const info = await this.lnrpc.listChannels();
+        return `Open channels:\n${JSON.stringify(info)}`;
+      }
+      default:
+        throw new Error('Question not recognized');
+    }
+  };
+
+  specifyChannelOpeningParameters = async () => {
+    const { peers } = await this.lnrpc.listPeers();
+    const { pubKey } = await inquirer.prompt<{ pubKey: string }>({
+      type: 'list',
+      name: 'pubKey',
+      message: 'Which peer would you like to open a channel with?',
+      choices: peers.map(e => e.pubKey),
```

Comment:
try as I might, I couldn't figure out how to list peers by alias (`getNodeInfo` doesn't seem to work for peers that aren't in your "graph" which I think means those without a valid payment path). Open to any feedback on how to display aliases

Reply:
You are correct. It's not possible to `getNodeInfo` for nodes that are not in your graph. Unfortunately, there isn't anything we can do about that. One suggestion is you could catch the error and use an ellipsed pubkey `02d3...f422` instead of the alias for nodes not in the graph.